### PR TITLE
[7.0] Remove add module exports internally

### PR DESCRIPTION
**Pull request by chicoxyzzy**
| Q                        | A 
| ------------------------ | ---
| Patch: Bug Fix?          | no
| Major: Breaking Change?  | yes
| Minor: New Feature?      | no
| Deprecations?            | yes
| Spec Compliancy?         | no
| Tests Added/Pass?        | no
| Fixed Tickets            | #5127 
| License                  | MIT
| Doc PR                   | no
| Dependency Changes       | no

### CODE REVIEW

File: packages/babel-core/register.js [hunk 1]
```diff
@@ -1,3 +1,3 @@
 /* eslint max-len: 0 */
 // TODO: eventually deprecate this console.trace("use the `babel-register` package instead of `babel-core/register`");
-module.exports = require("babel-register");
+module.exports = require("babel-register").default;
```
Comment (chicoxyzzy) [inline packages/babel-core/register.js]: Should we also deprecate this in other PR?

Comment (danez) [inline packages/babel-core/register.js]: Yeah good point, we definitely should do that. Can you please create an issue, or a PR if you have time? Thanks.

Comment (chicoxyzzy) [inline packages/babel-core/register.js]: #5132

File: packages/babel-core/test/transformation.js [hunk 2]
```diff
@@ -1 +1,3 @@
-require("babel-helper-transform-fixture-test-runner")(__dirname + "/fixtures/transformation", "transformation");
+import runner from "babel-helper-transform-fixture-test-runner";
+
+runner(`${__dirname}/fixtures/transformation`, "transformation");
```
Comment (xtuc) [inline packages/babel-core/test/transformation.js]: :+1:

**Comment by chicoxyzzy** [comment]
rebased to resolve conflicts

**Comment by hzoo** [comment]
I think we could make almost all changes on 6.x actually since its just changing to use import? Only other change is actually removing the plugin for 7.0 (just an idea)

**Comment by chicoxyzzy** [comment]
Great idea! I'll try to exclude plugin removing and run tests against master branch.

**Comment by hzoo** [comment]
Might be kind of annoying so don't have to (I don't know if there would be any merge conflicts otherwise from not doing it so we can just leave it as well)

**Comment by chicoxyzzy** [comment]
I can rebase this branch later, no problems! Just created #5138

**Comment by chicoxyzzy** [comment]
I think we should take change to 7.0 from master, then I'll rebase if there will be conflicts

**Comment by hzoo** [comment]
oops merged in the wrong order..

**Pull request merged**